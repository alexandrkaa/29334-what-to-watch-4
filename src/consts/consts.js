export const MovieCardFullTabsIds = {
  OVERVIEW: `OVERVIEW`,
  DETAILS: `DETAILS`,
  REVIEWS: `REVIEWS`,
};
export const PreviewVideoSize = {
  width: 280,
  height: 175
};

export const MovieCardFullTabs = [
  {
    id: `OVERVIEW`,
    name: `Overview`,
  },
  {
    id: `DETAILS`,
    name: `Details`
  },
  {
    id: `REVIEWS`,
    name: `Reviews`
  },
];

export const SECS_PER_HOUR = 3600;
export const MOVIES_LIKE_THIS_NUM = 4;
export const DEFAULT_GENRE = `All genres`;
export const MOVIES_LIMIT = 8;
export const GENRES_LIMIT = 8;
export const START_PREVIEW_DELAY = 1000;
export const BASE_URL = `https://htmlacademy-react-3.appspot.com`;
export const BASE_API_URL = `${BASE_URL}/wtw`;
export const API_REQUEST_TIMEOUT = 5000;

export const FieldsIds = {
  EMAIL_FIELD_ID: `user-email`,
  PASSWORD_FIELD_ID: `user-password`,
  RATING_FIELD_ID: `rating`,
  COMMENTS_FIELD_ID: `comments`,
};

export const SignInFields = [
  {
    id: FieldsIds.EMAIL_FIELD_ID,
    label: `Email address`,
    type: `email`,
    placeholder: `Email address`,
  },
  {
    id: FieldsIds.PASSWORD_FIELD_ID,
    label: `Password`,
    type: `password`,
    placeholder: `Password`,
  },
];

export const NetworkErrors = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};

export const AppRoutes = {
  MAIN_PAGE: `/`,
  FULL_PLAYER_PAGE: `/player`,
  FILM_PAGE: `/films`,
  LOGIN_PAGE: `/login`,
  REVIEW_PAGE: `/review`,
  USERFAVORITE_PAGE: `/mylist`,
};

export const REVIEW_STARS_NUMBER = 5;

export const ReviewTextLength = {
  MIN: 50,
  MAX: 400,
};

export const RatingScores = {
  AWESOME: {
    score: 10,
    name: `Awesome`,
  },
  VERYGOOD: {
    score: 8,
    name: `Very good`,
  },
  GOOD: {
    score: 5,
    name: `Good`,
  },
  NORMAL: {
    score: 3,
    name: `Normal`,
  },
  BAD: {
    score: 0,
    name: `Bad`,
  },
};
