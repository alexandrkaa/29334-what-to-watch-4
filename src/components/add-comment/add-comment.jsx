import React from 'react';
import Header from '../header/header.jsx';
import withMovie from '../../hocs/with-movie/with-movie.js';
import {movieType} from '../../types/types.js';
import Breadcrumbs from '../breadcrumbs/breadcrumbs.jsx';
import AddCommentForm from '../add-comment-form/add-comment-form.jsx';
import UserProfile from '../user-profile/user-profile.jsx';
import {ComponentsKeys} from '../../consts/consts.js';

const AddComment = (props) => {
  const {
    id: movieId,
    title,
    movieImage,
    movieBackground,
  } = props.movie;

  return (
    <section className="movie-card movie-card--full">
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={movieBackground} alt={title} />
        </div>

        <h1 className="visually-hidden">WTW</h1>
        <Header>
          <Breadcrumbs key={ComponentsKeys.BREADCRUMBS} movie={props.movie} />
          <UserProfile key={ComponentsKeys.USERPROFILE} />
        </Header>

        <div className="movie-card__poster movie-card__poster--small">
          <img src={movieImage} alt="The Grand Budapest Hotel poster" width="218" height="327" />
        </div>
      </div>
      <AddCommentForm movieId={movieId} />
    </section>

  );
};

AddComment.propTypes = {
  movie: movieType.isRequired,
};

export {AddComment};
export default withMovie(AddComment);
