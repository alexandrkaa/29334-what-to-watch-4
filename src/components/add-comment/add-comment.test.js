import React from 'react';
import renderer from 'react-test-renderer';
import {AddComment} from './add-comment.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {MOVIES_LIMIT} from '../../consts/consts.js';
import {BrowserRouter} from 'react-router-dom';
import createAPI from '../../api/api.js';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';

const api = createAPI(() => {});
const apiMock = new MockAdapter(api);
apiMock
      .onGet(`/login`)
      .reply(200, {});

const middlewares = [thunk.withExtraArgument(apiMock)];

const mockStore = configureStore(middlewares);

const mockData = {
  titleMovie: {
    id: 4,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    movieDate: `2000`,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
    movieDirector: `Wes Anderson`,
    movieGenre: `Thriller`,
    movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
    movieRatingCount: `349 ratings`,
    movieRatingLevel: `Bad`,
    movieRatingScore: `3`,
    movieStarring: [`Jude Law`, `Willem Dafoe`],
    title: `Dardjeeling Limited`,
    moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#000000`,
    movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  },

  moviesList: [
    {
      id: 4,
      image: `img/we-need-to-talk-about-kevin.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2000`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Wes Anderson`,
      movieGenre: `Thriller`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `349 ratings`,
      movieRatingLevel: `Bad`,
      movieRatingScore: `3`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Dardjeeling Limited`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    },
    {
      id: 3,
      image: `img/aviator.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2008`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Christopher Nolan`,
      movieGenre: `Kids & Family`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `272 ratings`,
      movieRatingLevel: `Normal`,
      movieRatingScore: `2`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Johnny English`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    }
  ],
  movieGenres: [
    `Thriller`,
    `Kids & Family`
  ]
};

it.skip(`<AddComment /> should match snapshot without error`, () => {
  const {titleMovie, moviesList, movieGenres} = mockData;
  const store = mockStore({
    MOVIE_DATA: {
      titleMovie,
      moviesList,
      movieGenres,
      loadingMovies: false,
      loadingMoviesError: false,
    },
    COMMENTS_DATA: {
      loadingComments: false,
      moviesComments: [],
      postCommentInProgress: false,
      postCommentError: false,
    },
    MOVIE: {
      activeGenre: `All genres`,
      moviesRenderLimit: MOVIES_LIMIT,
    },
    USER: {
      authorizationStatus: `AUTH`,
      userData: {
        id: 1,
        email: `qwe@qwe.ru`,
        name: `qwe`,
        avatarUrl: `/img/avatar.jpg`
      }
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <AddComment
              movieId={1}
              movie={mockData.moviesList[0]}
            />
          </BrowserRouter>
        </Provider>
    )
    .toTree();
  expect(tree).toMatchSnapshot();
});
