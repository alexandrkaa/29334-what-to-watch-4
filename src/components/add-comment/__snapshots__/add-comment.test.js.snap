// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddComment /> should match snapshot without error 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Dardjeeling Limited"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <Memo(Header)>
      <Breadcrumbs
        movie={
          Object {
            "backgroundColor": "#000000",
            "id": 4,
            "image": "img/we-need-to-talk-about-kevin.jpg",
            "isFavorite": false,
            "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
            "movieDate": "2000",
            "movieDescription": "In the 1930s, the Grand Budapest Hotel is a populaâ€¦y boy, becomes Gustave&apos;s friend and protege.",
            "movieDirector": "Wes Anderson",
            "movieGenre": "Thriller",
            "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
            "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "movieRatingCount": "349 ratings",
            "movieRatingLevel": "Bad",
            "movieRatingScore": "3",
            "movieStarring": Array [
              "Jude Law",
              "Willem Dafoe",
            ],
            "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "title": "Dardjeeling Limited",
          }
        }
      />
      <Memo(Connect(UserProfile)) />
    </Memo(Header)>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <Memo(Connect(WithCommentHOC))
    movieId={4}
  />
</section>
`;
